go/geth_adapter/adapter_test.go

```diff
@@ -75,3 +86,311 @@ func TestRunContextAdapter_SetBalanceHasCorrectEffect(t *testing.T) {
 		})
 	}
 }
+
+func TestRunContextAdapter_SetAndGetNonce(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+	adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+
+	address := tosca.Address{0x42}
+	nonce := uint64(123)
+
+	stateDb.EXPECT().SetNonce(gc.Address(address), nonce)
+	adapter.SetNonce(address, nonce)
+
+	stateDb.EXPECT().GetNonce(gc.Address(address)).Return(nonce)
+	adapter.GetNonce(address)
+}
+
+func TestRunContextAdapter_SetAndGetCode(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+	adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+
+	address := tosca.Address{0x42}
+	code := []byte{1, 2, 3}
+
+	stateDb.EXPECT().SetCode(gc.Address(address), code)
+	adapter.SetCode(address, code)
+
+	stateDb.EXPECT().GetCode(gc.Address(address)).Return(code)
+	adapter.GetCode(address)
+}
+
+func TestRunContextAdapter_SetAndGetStorage(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+	adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+
+	address := tosca.Address{0x42}
+	key := tosca.Key{10}
+	original := tosca.Word{0}
+	current := tosca.Word{1}
+	future := tosca.Word{2}
+
+	stateDb.EXPECT().GetState(gc.Address(address), gc.Hash(key)).Return(gc.Hash(current))
+	stateDb.EXPECT().GetCommittedState(gc.Address(address), gc.Hash(key)).Return(gc.Hash(original))
+	stateDb.EXPECT().SetState(gc.Address(address), gc.Hash(key), gc.Hash(future))
+	adapter.SetStorage(address, key, future)
```

Comment:
todo: check the output of the `SetStorage` function (for this one example should be enough since the implementation uses `GetStorageStatus` internally, which is tested independently)